pkg/queue/manager.go

```diff
@@ -452,18 +453,44 @@ func (m *Manager) QueueInadmissibleWorkloads(ctx context.Context, cqNames sets.S
 // 2. add events of any cluster queue in the cohort.
 // 3. update events of any cluster queue in the cohort.
 // 4. update of cohort.
+//
+// WARNING: must hold a read-lock on the manager when calling,
+// or otherwise risk encountering an infinite loop if a Cohort
+// cycle is introduced.
 func (m *Manager) moveWorkloadsCQ(ctx context.Context, cq *ClusterQueue) bool {
 	if cq.HasParent() {
 		return m.moveWorkloadsCohort(ctx, cq.Parent())
 	}
 	return cq.QueueInadmissibleWorkloads(ctx, m.client)
 }
 
+// moveWorkloadsCohorts checks for a cycle, the moves all inadmissible
+// workloads in the Cohort tree. If a cycle exists, or no workloads were
+// moved, it returns false.
+//
+// WARNING: must hold a read-lock on the manager when calling,
+// or otherwise risk encountering an infinite loop if a Cohort
+// cycle is introduced.
 func (m *Manager) moveWorkloadsCohort(ctx context.Context, cohort *cohort) bool {
```

Comment:
```suggestion
func (m *Manager) queueInadmissibleWorkloadsForCohort(ctx context.Context, cohort *cohort) bool {
```
non-blocking nit (as this is pre-existing name), but I think "moveWorkloadsCohort" is too ambigous

Reply:
Decided in another PR to keep name short - it was hard to keep name short and unambiguous, as the function for CQs does almost the same thing, but starting at the CQ

See https://github.com/kubernetes-sigs/kueue/pull/2971#discussion_r1745396750

Reply:
Renamed slightly - `s/move/requeue/`

Reply:
sgtm